pkg/binary/envoy/debug/pii.go

```diff
@@ -0,0 +1,72 @@
+package debug
+
+import (
+	"fmt"
+	"bitbucket.org/creachadair/shell"
+	"crypto/sha256"
+)
+
+// filter the array of logs using formatStr and validFieldNames, hash the resulting array of logs and return the final array
+func process(logsArr []string, formatStr string, validFieldNames map[string]bool) (processedLogs []string) {
+	defer func() {
+		if r := recover(); r != nil {
+			fmt.Println("process func panicking, recovering")
+		}
+	} ()
+
+	fieldNames, ok := shell.Split(formatStr)
+
+	if !ok {
+		panic(fmt.Sprint("error in splitting format string: %s", formatStr))
+	}
+
+	testForValidLog :=  func(log string) bool {
+		fields, ok := shell.Split(log)
+		if !ok {
+			panic(fmt.Sprint("error in splitting log: %s", log))
+		}
+		// TODO: consider using regex to check fields instead of merely verifiying the number of fields
+		return len(fields) == len(fieldNames)
+	}
+	validLogs := filter(logsArr, testForValidLog)
+
+	// filter and hash each log in the array of valid logs with relevant fields
+	for i,_ := range(validLogs) {
```

Comment:
`range` is not a function: `for i, _ := range validLogs { ... }`

Reply:
The reason golang uses the `for index, element := range array` scheme is to avoid `array[index]` everywhere in the body of the for loop. In this case, you should update the loop like `for i, log := range logs` and replace all instances of `validLogs[i]` with `log`. In fact, I think you'll find that if you do this, you won't even need to declare the `i` variable at all, and you can just `for _, log := range logs { ... }`